Python/Include/Autobox.hpp

```diff
@@ -18,8 +18,11 @@ namespace Rover::Details {
         : m_obj(std::move(obj)) {}
 
       Type generate(Evaluator& e) {
-        auto wrapper = std::shared_ptr<Evaluator>(&e, [](auto ptr) {});
-        return m_obj.attr("generate")(std::move(wrapper)).cast<Type>();
+        auto evaluator_ptr = std::shared_ptr<Evaluator>(&e, [](auto) {});
```

Comment:
This took me a while to figure out. `pybind11::cast` will take ownership of the C++ object, and the garbage collector will delete it even if I pass a pointer with an empty deleter, no matter the `return_value_policy`. So I have to get a `pybind11::handle` (no ownership) out of the C++ object, and then use `reinterpret_borrow `to create a non-owning `pybind11::object`. Unfortunately, there seems to be no way to construct a `pybind11::handle` directly using the public API, so I have to create an owning `pybind11::object` and release it first, which will produce the handle while circumventing the destructor.